front/src/modules/infra/components/InfraSelector/InfraSelectorModalBodyStandard.tsx

```diff
@@ -49,6 +54,20 @@ export default function InfraSelectorModalBodyStandard({
   const { closeModal } = useContext(ModalContext);
   const navigate = useNavigate();
 
+  const { data: subjectsList } = mockedEditoastApi.endpoints.getSubjects.useQuery();
+
+  const { data: resourceGrants } =
+    mockedEditoastApi.endpoints.getGrantsByResourceType.useQuery('infra');
+
+  const payload = useMemo(() => ({ infra: infrasList.map((infra) => infra.id) }), [infrasList]);
+
+  const userResourcesGrants = useResourcesGrants(payload);
+
+  const userGrantByInfra = useMemo(
```

Comment:
The type here is Map<number (infraId), { id: number, grant: Grant }>. It could be simpler, only Map<number (infraId), Grant>